### PR TITLE
Fix method lookup for single char class names

### PR DESCRIPTION
**Pull request by caspiano**
While looking at #12672, I noticed that single-character namespaces were not matched when parsing method names from code blocks

### CODE REVIEW

**Comment by caspiano** [comment]
This will probably have conflicts after #12683 is merged, but thought it would be better to have things separated.

**Review by oprypin — APPROVED** [review]

File: src/compiler/crystal/tools/doc/markd_doc_renderer.cr [hunk 1]
```diff
@@ -74,9 +74,9 @@ class Crystal::Doc::MarkdDocRenderer < Markd::HTMLRenderer
 
     # Check Type#method(...) or Type or #method(...)
     text.gsub %r(
-      ((?:\B::)?\b[A-Z]\w+(?:\:\:[A-Z]\w+)*|\B|(?<=\bself))([#.])([\w<=>+\-*\/\[\]&|?!^~]+[?!]?)(?:\((.*?)\))?
+      ((?:\B::)?\b[A-Z]\w*(?:\:\:[A-Z]\w*)*|\B|(?<=\bself))([#.])([\w<=>+\-*\/\[\]&|?!^~]+[?!]?)(?:\((.*?)\))?
         |
-      ((?:\B::)?\b[A-Z]\w+(?:\:\:[A-Z]\w+)*)
+      ((?:\B::)?\b[A-Z]\w*(?:\:\:[A-Z]\w*)*)
```
Comment (straight-shoota) [inline src/compiler/crystal/tools/doc/markd_doc_renderer.cr:79]: I don't think I fully comprehend the intention of this regex, neither before nor after this change.
What's the purpose of matching for double colon followed by a letter and (now optional) whitespace? 🤔

**Comment by HertzDevil** [comment]
I wonder how much overhead there'd be if we use a full parser here.

File: src/compiler/crystal/tools/doc/markd_doc_renderer.cr (hunk 1 above)
Comment (oprypin) [reply src/compiler/crystal/tools/doc/markd_doc_renderer.cr:79]: I think maybe the main confusion is that `\w` is "word character", not "whitespace" as I'm guessing that you assumed 😅

**Comment by caspiano** [comment]
> I wonder how much overhead there'd be if we use a full parser here.

I think after the two changes I made it's... 'good enough' as far as I can tell. A refactor wouldn't hurt, but might be a bit much effort to expend.
There's a few pending specs I can try to pick up in the renderer related to method matching.

File: src/compiler/crystal/tools/doc/markd_doc_renderer.cr (hunk 1 above)
Comment (straight-shoota) [reply src/compiler/crystal/tools/doc/markd_doc_renderer.cr:79]: Yep. Doh 🤦

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**